gcloud/bigquery/job.py

```diff
@@ -0,0 +1,702 @@
+        """
+        if not all(isinstance(field, SchemaField) for field in value):
+            raise ValueError('Schema items must be fields')
+        self._schema = tuple(value)
+
+    @property
+    def etag(self):
+        """ETag for the job resource.
+
+        :rtype: string, or ``NoneType``
+        :returns: the ETag (None until set from the server).
+        """
+        return self._properties.get('etag')
+
+    @property
+    def job_id(self):
+        """ID for the job resource.
+
+        :rtype: string, or ``NoneType``
+        :returns: the ID (None until set from the server).
+        """
+        return self._properties.get('id')
+
+    @property
+    def self_link(self):
+        """URL for the job resource.
+
+        :rtype: string, or ``NoneType``
+        :returns: the URL (None until set from the server).
+        """
+        return self._properties.get('selfLink')
+
+    @property
+    def user_email(self):
+        """E-mail address of user who submitted the job.
+
+        :rtype: string, or ``NoneType``
+        :returns: the URL (None until set from the server).
+        """
+        return self._properties.get('user_email')
+
+    @property
+    def created(self):
+        """Datetime at which the job was created.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        :returns: the creation time (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            millis = statistics.get('creationTime')
+            if millis is not None:
+                return _datetime_from_microseconds(millis * 1000.0)
+
+    @property
+    def started(self):
+        """Datetime at which the job was started.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        :returns: the start time (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            millis = statistics.get('startTime')
+            if millis is not None:
+                return _datetime_from_microseconds(millis * 1000.0)
+
+    @property
+    def ended(self):
+        """Datetime at which the job finished.
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
+        :returns: the end time (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            millis = statistics.get('endTime')
+            if millis is not None:
+                return _datetime_from_microseconds(millis * 1000.0)
+
+    @property
+    def input_file_bytes(self):
+        """Count of bytes loaded from source files.
+
+        :rtype: integer, or ``NoneType``
+        :returns: the count (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            return int(statistics['load']['inputFileBytes'])
```

Comment:
The keys above can all be missing at any point.  The statistics will either be present or not, AFAICT.